libbeat/common/file/interval_rotator.go

```diff
@@ -93,7 +96,41 @@ func (r *intervalRotator) initialize() error {
 			return lastInterval != currentInterval
 		}
 	}
-	return nil
+
+	// if rotation is not allowed on startup, find the newest rotated file last modification time
+	// or that of the unrotated log file
+	if !rotateOnStartup {
+		logsDir := filepath.Dir(filename)
+		logfiles, err := ioutil.ReadDir(logsDir)
+		if err != nil && log != nil {
+			log.Debugw("Not attempting to find last rotated time, configured logs dir cannot be opened: %v", err)
+			return
+		}
+
+		r.lastRotate = time.Time{}
+		if len(logfiles) == 1 && logfiles[0].Name() == filename {
+			if log != nil {
+				log.Debugw("Setting last rotated time to the last modification time of the log")
+			}
+
+			r.lastRotate = logfiles[0].ModTime()
+			return
+		}
+		basenamePrefix := filepath.Base(filename) + "-"
```

Comment:
This is somewhat of a nitpick: before this PR the delimiter used in the interval-rotated filenames was being used in a single method in the code:  https://github.com/elastic/beats/blob/871bea3bce2b7db17a57d87573c5601ae131ae1b/libbeat/common/file/interval_rotator.go#L136-L154.

With this PR the delimiter is now being used in a couple of methods, introducing [Connascence of Name](https://en.wikipedia.org/wiki/Connascence#Connascence_of_Name_(CoN)). Maybe extract it into a const and update all places to use it?

Reply:
Well, I tried this change. IMO it makes the format strings unreadable.

Reply:
Fair enough, thanks for trying it.